File: core-scala/src/main/java/io/asterisque/core/session/NoSuchServiceException.java
package io.asterisque.core.session;

import io.asterisque.wire.ProtocolException;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * ピアの指定したサービスが見つからないときに発生する例外です。
 */
public class NoSuchServiceException extends ProtocolException {
  public NoSuchServiceException(@Nonnull String message) {
    super(message);
  }

  public NoSuchServiceException(@Nonnull String message, @Nullable Throwable ex) {
    super(message, ex);
  }
}

File: core-scala/src/main/scala/io/asterisque/wire/rpc/NoSuchServiceException.scala
package io.asterisque.wire.rpc

import io.asterisque.wire.ProtocolException
import javax.annotation.{Nonnull, Nullable}

class NoSuchServiceException(@Nullable message:String, @Nullable ex:Throwable) extends ProtocolException(message, ex) {
  def this(@Nonnull message:String) = this(message, null)

  def this(@Nonnull ex:Throwable) = this("", ex)

  def this() = this("", null)
}
